### ISSUE TITLE
Bug in m.set_signal_range

### ISSUE THREAD

**Issue by LMSC-NTappy** [hyperspy/hyperspy#2457]
I realized this while working on #2454

**Bug description**
When called with 'None', `m.set_signal_range` crops the last pixel of the axis. The situation is encountered when eelsmodel fit_background is called without argument and there are no active edges in the model. See below
https://github.com/hyperspy/hyperspy/blob/30d28d23f074bbccc92ff6a537addb22566d2d1d/hyperspy/tests/model/test_eelsmodel.py#L218-L223
the test asserts 2.13567839196 when `s.mean(-1)` evaluates to 2.14

This is because `value_range_to_indices(x1, x2)` evaluates to (0,len(axis)-1) when input with None

**Minimal Example to reproduce bug**
```python
s = hs.signals.Signal1D(np.ones(100))
m = s.create_model()
m.set_signal_range(None,None)
m.channel_switches
#array([ True,  True,  True,  True,  True,  True,  True,  True,  True, False])
```

Probably not a big deal but thought I would share

**Comment by francisco-dlp** [comment]
Good catch, that's an ancient bug!

**Comment by ericpre** [comment]
Should be fixed in #2509.
